### PR TITLE
RavenDB-18445 initial impl. for sharded get ongoing tasks

### PR DESCRIPTION
**Pull request by karmeli87**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18445

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks;
+
+internal class ShardedOngoingTasksHandlerProcessorForGetOngoingTasks : AbstractOngoingTasksHandlerProcessorForGetOngoingTasks<TransactionOperationContext>
+{
+    [NotNull]
+    private readonly ShardedDatabaseRequestHandler _handler;
+
+    public ShardedOngoingTasksHandlerProcessorForGetOngoingTasks([NotNull] ShardedDatabaseRequestHandler requestHandler, [NotNull] JsonContextPoolBase<TransactionOperationContext> contextPool) : base(requestHandler, contextPool)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs]: better to remove context pool and passs requestHandler.ContextPool IMO

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+using Raven.Client.Json.Serialization;
+using Raven.Client.ServerWide;
+using Raven.Server.Documents.Sharding.Handlers;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web.System.Processors.OngoingTasks;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Handlers.Processors.OngoingTasks;
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs]: This should be moved to a different namespace, we do have one for those handlers

File: src/Raven.Server/Web/RequestHandler.Helper.cs [hunk 3]
```diff
@@ -257,7 +257,7 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, stream))
             }
         }
 
-        internal async void InternalGetState(OperationState state, JsonOperationContext context)
+        internal async Task InternalGetState(OperationState state, JsonOperationContext context)
```
Comment (ppekrol) [inline src/Raven.Server/Web/RequestHandler.Helper.cs]: Nice

Comment (ppekrol) [reply src/Raven.Server/Web/RequestHandler.Helper.cs]: rename to InternalGetStateAsync

File: src/Raven.Server/Web/System/Processors/OngoingTasks/OngoingTasksHandlerProcessorForGetOngoingTasks.cs [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+namespace Raven.Server.Web.System.Processors.OngoingTasks;
+
+internal class OngoingTasksHandlerProcessorForGetOngoingTasks : AbstractOngoingTasksHandlerProcessorForGetOngoingTasks<DocumentsOperationContext>
+{
+    [NotNull]
+    private readonly DatabaseRequestHandler _requestHandler;
+    private readonly DocumentDatabase _database;
+    private readonly ServerStore _server;
+
+    public OngoingTasksHandlerProcessorForGetOngoingTasks([NotNull] DatabaseRequestHandler requestHandler, [NotNull] JsonContextPoolBase<DocumentsOperationContext> contextPool) : base(requestHandler, contextPool)
```
Comment (ppekrol) [inline src/Raven.Server/Web/System/Processors/OngoingTasks/OngoingTasksHandlerProcessorForGetOngoingTasks.cs]: can probably use the pool from requestHandler.ContextPool

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs]: done

File: src/Raven.Server/Web/RequestHandler.Helper.cs (hunk 3 above)
Comment (karmeli87) [reply src/Raven.Server/Web/RequestHandler.Helper.cs]: done

File: src/Raven.Server/Web/System/Processors/OngoingTasks/OngoingTasksHandlerProcessorForGetOngoingTasks.cs (hunk 4 above)
Comment (karmeli87) [reply src/Raven.Server/Web/System/Processors/OngoingTasks/OngoingTasksHandlerProcessorForGetOngoingTasks.cs]: done

File: src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Handlers/Processors/OngoingTasks/ShardedOngoingTasksHandlerProcessorForGetOngoingTasks.cs]: done

**Pull request merged**